tools/builders/util/copy-directory.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * @license
+ * Copyright 2019 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { statSync, existsSync, promises as fs } from 'fs';
+import { join } from 'path';
+
+/** Asynchronously copies a directory and all its content to the target directory recursively */
+export async function copyDirectory(
+  source: string,
+  target: string,
+): Promise<void> {
+  const exists = existsSync(source);
+  if (exists) {
+    const stats = statSync(source);
+    const isDirectory = exists && stats.isDirectory();
```

Comment:
Exists check is not necessary, as it has to be true to reach this branch of code in the first place. You can remove line 28 and change line 29 to: 
```ts
if (stats.isDirectory()) {
```